ast/environment.go

```diff
@@ -293,6 +293,7 @@ func parseField(name string, dest reflect.Value, node syntax.Node) syntax.Diagno
 		xv := reflect.ValueOf(x)
 		if !xv.Type().AssignableTo(dest.Type()) {
 			diags.Extend(exprFieldTypeMismatchError(name, dest.Interface(), x))
+			return diags
```

Comment:
The only concern I have about an early return here is that it leaves `dest` uninitialized. Instead, we might want to create a new zero value of the destination type and fall through. This would be consistent with the other branches.

I think that's just `v = reflect.New(dest.Type().Elem())`.

Reply:
Updated